### PR TITLE
Run executing events + bug fixes 

### PR DESCRIPTION
**Pull request by AydinHassan**

### CODE REVIEW

File: src/ExerciseRunner/CgiRunner.php [hunk 1]
```diff
@@ -202,7 +202,9 @@ public function run($fileName, OutputInterface $output)
 
             $output->writeTitle("Output");
             $output->emptyLine();
-            $process->run(function ($outputType, $outputBuffer) use ($output) {
+            $process->start();
+            $this->eventDispatcher->dispatch(new CgiExecuteEvent('cgi.run.executing', $request, ['output' => $output]));
+            $process->wait(function ($outputType, $outputBuffer) use ($output) {
```
Comment (mikeymike) [inline src/ExerciseRunner/CgiRunner.php]: Nice!

File: src/ExerciseRunner/CliRunner.php [hunk 2]
```diff
@@ -145,7 +145,9 @@ public function run($fileName, OutputInterface $output)
 
         $output->writeTitle("\nOutput");
```
Comment (mikeymike) [inline src/ExerciseRunner/CliRunner.php]: fancy taking the `\n` out while your here and making the last write (L143) in the block above a writeLine instead ?

Comment (mikeymike) [inline src/ExerciseRunner/CliRunner.php]: part fixes the issue #77

Comment (AydinHassan) [inline src/ExerciseRunner/CliRunner.php]: sure

**Comment by mikeymike** [comment]
just fixed the test for this, force pushed it though :speak_no_evil: :P

**Comment by mikeymike** [comment]
Gna merge this, I broke the tests in the last merge :100:

**Pull request merged**